src/agora/dobeamweights.cc

```diff
@@ -389,18 +389,32 @@ static inline void TransposeGather(size_t cur_sc_id, float* src, float*& dst,
 }
 
 void DoBeamWeights::ComputeBeams(size_t tag) {
+  //Request was generated from gen_tag_t::FrmSc
   const size_t frame_id = gen_tag_t(tag).frame_id_;
   const size_t base_sc_id = gen_tag_t(tag).sc_id_;
   const size_t frame_slot = frame_id % kFrameWnd;
   if (kDebugPrintInTask) {
     std::printf("In doZF thread %d: frame: %zu, base subcarrier: %zu\n", tid_,
                 frame_id, base_sc_id);
   }
-  const size_t num_subcarriers =
-      std::min(cfg_->BeamBlockActiveSc(), cfg_->OfdmDataNum() - base_sc_id);
+
+  //Process BeamBlockSize (or less) number of carriers
+  //cfg_->OfdmDataNum() is the total number of usable subcarriers
+  size_t num_subcarriers =
+      std::min(cfg_->BeamBlockSize(), cfg_->OfdmDataNum() - base_sc_id);
```

Comment:
num_subcarriers is better named as "sc_range" since the number of processed subcarriers could be different.
So we will have three variables to control the loop:
sc_start (renamed from start_sc)
sc_range
sc_inc